pkg/repository/v1/shared.go

```diff
@@ -25,17 +27,19 @@ type taskExternalIdTenantIdTuple struct {
 }
 
 type sharedRepository struct {
-	pool                      *pgxpool.Pool
-	v                         validator.Validator
-	l                         *zerolog.Logger
-	queries                   *sqlcv1.Queries
-	queueCache                *cache.Cache
-	stepExpressionCache       *cache.Cache
-	tenantIdWorkflowNameCache *cache.Cache
-	celParser                 *cel.CELParser
-	env                       *celgo.Env
-	taskLookupCache           *lru.Cache[taskExternalIdTenantIdTuple, *sqlcv1.FlattenExternalIdsRow]
-	m                         *metered.Metered
+	pool                        *pgxpool.Pool
+	v                           validator.Validator
+	l                           *zerolog.Logger
+	queries                     *sqlcv1.Queries
+	queueCache                  *cache.Cache
+	stepExpressionCache         *cache.Cache
+	tenantIdWorkflowNameCache   *expirable.LRU[string, *sqlcv1.ListWorkflowsByNamesRow]
+	stepsInWorkflowVersionCache *expirable.LRU[string, []*sqlcv1.ListStepsByWorkflowVersionIdsRow]
+	stepIdLabelsCache           *expirable.LRU[string, []*sqlcv1.GetDesiredLabelsRow]
```

Comment:
I would like to start moving towards usage of the `expirable` package instead of our home-built cache, because of nicer typing/LRU support - I think we can fully make this transition after we deprecate v0.